src/PostgREST/QueryBuilder.hs

```diff
@@ -498,6 +500,9 @@ insertableValue :: JSON.Value -> SqlFragment
 insertableValue JSON.Null = "null"
 insertableValue v = (<> "::unknown") . pgFmtLit $ unquoted v
 
+insertableValueWithType :: Text -> JSON.Value -> SqlFragment
+insertableValueWithType t v = (<> "::" <> t) . pgFmtLit $ unquoted v
```

Comment:
Done.  BTW, I didn't prefix the key with `postgres-` to remain consistent with all other `Prefer` keys.